Sources/PayPalDataCollector/PayPalDataCollector.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import PPRiskMagnes
+#if canImport(PaymentsCore)
+import PaymentsCore
+#endif
+
+public class PayPalDataCollector {
+
+    // MARK: - Properties
+    private let config: CoreConfig
+    private let magnesSDK: MagnesSDKProtocol
+    private let deviceInspector: DeviceInspectorProtocol
+
+    // MARK: - Initializers
+    public convenience init(config: CoreConfig) {
+        self.init(config: config, magnesSDK: MagnesSDK.shared(), deviceInspector: DeviceInspector())
+    }
+
+    init(config: CoreConfig, magnesSDK: MagnesSDKProtocol, deviceInspector: DeviceInspectorProtocol) {
+        self.config = config
+        self.magnesSDK = magnesSDK
+        self.deviceInspector = deviceInspector
+    }
+
+    // MARK: - Computed Properties
+    private var magnesEnvironment: MagnesSDK.Environment {
+        switch config.environment {
+        case .sandbox:
+            return .SANDBOX
+        case .production:
+            return .LIVE
+        }
+    }
+
+    // MARK: - PayPalDataCollector
+    public func collectDeviceData(additionalData: [String: String] = [:]) -> String {
+        let deviceIdentifier = deviceInspector.payPalDeviceIdentifier()
+        let params = MagnesSetupParams(
+            env: magnesEnvironment,
+            appGuid: deviceIdentifier,
+            isRemoteConfigDisabled: false,
+            isBeaconDisabled: false,
+            source: .PAYPAL
+        )
+        try? magnesSDK.setUpWithParams(params)
+
+        let result = try? magnesSDK.collectDeviceData(withPayPalClientMetadataId: "", withAdditionalData: additionalData)
+        let clientMetadataId = result?.getPayPalClientMetaDataId() ?? ""
+
+        return """
+            {"correlation_id":"\(clientMetadataId)"}
+        """.trimmingCharacters(in: .whitespacesAndNewlines)
```

Comment:
Could we simplify this by doing `return "{"correlation_id":"\(clientMetadataId)"}".trimmingCharacters(in: .whitespacesAndNewlines)`?

Reply:
Yeah, I think we just escape the quotes in the BT SDK. I think since it's only 1 param the readability is still pretty high. Also curious why one line single quotes broke the build, that's weird for sure!